### ISSUE TITLE
Request: be able to specify the location of metadata.yaml (and the rest of the charm code)

### ISSUE THREAD

**Issue by xavpaice** [canonical/charmcraft#644]
In repositories such as https://git.launchpad.net/charm-grafana for reactive charms, it is possible to charmcraft pack the charm using:

```yaml
type: charm
parts:
  charm:
    source: src
    plugin: reactive
    build-snaps: [charm]
bases:
    - name: ubuntu
      channel: "20.04"
      architectures: ["amd64"]
```

Note that `source: src` is specified because we put the charm in a `src` directory.  When we build the charm using reactive tools we run `charm build src/`.  Currently, in order to use charmcraft and the recipe in Launchpad "create charm recipe" I need to have charmcraft.yaml in the root of the repo, and that requires me to symlink metadata.yaml to the root also so that charmcraft can find it.

I'd like to be able to specify in charmcraft.yaml where the metadata.yaml is to be found - I'd have expected it to be located in the `source:` directive of the reactive plugin.

**Comment by xavpaice** [comment]
FWIW, I see that there's a `--from` argument available for `charmcraft build` which is great, but I can't add the CLI option to Launchpad - what I'm asking for is the ability to add something similar to charmcraft.yaml so we have charmcraft.yaml in the root of the repo, and the rest of the charm (including metadata.yaml) in a subdirectory.

**Comment by lengau** [comment]
This will no longer be required for reactive charms with charmcraft 3.0. (Fixed in #1524)

Rather than specifying the location of `metadata.yaml`, this uses the `metadata.yaml` generated by the `charm` tool.
